File: cpp/DeviceDriverTest.cpp
#include "DeviceDriver.hpp"
#include "MockMemoryDevice.hpp"
#include <gtest/gtest.h>

using namespace ::testing;

TEST(DeviceDriver, read)
{
    MockMemoryDevice *hardware = new MockMemoryDevice();
    DeviceDriver *device = new DeviceDriver(*hardware);
    ASSERT_EQ(0, device->read(0xFF));
}

File: CSharp/DeviceDriverTest.cs
using Xunit;

namespace device
{
    public class DeviceDriverTest
    {

        [Fact]
        public void Read_From_Hardware()
        {
            //TODO: replace the hardware with a test double
            IFlashMemoryDevice hardware = null;
            DeviceDriver driver = new DeviceDriver(hardware);
            byte data = driver.Read(0xFF);
            Assert.Equal(0b0, data);
        }

    }

}
